src/amplify-migration/githubUtils.ts

```diff
@@ -0,0 +1,63 @@
+import { AmplifyAppInfo } from "./amplifyUtils";
+import fs from "fs";
+import simpleGit from "simple-git";
+import { BRANCH_NAME, ORGANIZATION_NAME } from "./constants";
+import { Octokit } from "@octokit/core";
+
+export async function pushChangesToRemote({ repoPath }: AmplifyAppInfo) {
+  await simpleGit(repoPath).checkout("staging");
+  await simpleGit(repoPath).merge([BRANCH_NAME]);
+
+  await simpleGit(repoPath).push("origin", "staging");
+  await simpleGit(repoPath).deleteLocalBranch(BRANCH_NAME);
+  console.info("Merge and delete of add-trailing-slash branch successful");
+}
+export async function checkoutBranch(repoPath: string, repoName: string) {
+  if (fs.existsSync(repoPath)) {
+    console.info(
+      `Repository ${repoName} already exists. Pulling changes from origin.`
+    );
+    await simpleGit(repoPath).pull("origin", "staging");
+  } else {
+    console.info(`Cloning ${repoName} repository from ${ORGANIZATION_NAME}...`);
+    await simpleGit().clone(
+      `https://github.com/${ORGANIZATION_NAME}/${repoName}.git`,
+      repoPath
+    );
+  }
+
+  /**
+   * get the list of branches, guaranteed to be in
+   * local since we don't intend to push BRANCH_NAME to remote
+   */
+  const branches = await simpleGit(repoPath).branchLocal();
+  if (branches.all.includes(BRANCH_NAME)) {
+    console.log("Branch already exists. Checking out branch.");
+    await simpleGit(repoPath).checkout(BRANCH_NAME);
+  } else {
+    await simpleGit(repoPath).checkoutLocalBranch(BRANCH_NAME);
+  }
+}
+export async function isRepoEmpty(repoName: string): Promise<boolean> {
+  const octokit = new Octokit({
+    auth: process.env.GITHUB_ACCESS_TOKEN,
+  });
+  try {
+    const result = await octokit.request(
+      `GET /repos/${ORGANIZATION_NAME}/${repoName}/contents/README.md`,
+      {
+        owner: ORGANIZATION_NAME,
+        repo: repoName,
+        path: `README.md`,
```

Comment:
Do all repos have a README? Why is this file the indication of an empty/non-empty repo?

Reply:
Yes, our scripts (during site creation) adds a `README` for the repo, so the lack of this is suggestive of an empty repo